src/Bicep.Core/Analyzers/Linter/Rules/WhatIfShortCircuitingRule.cs

```diff
@@ -0,0 +1,161 @@
+        { }
+
+        public override string FormatMessage(params object[] values)
+        {
+            return string.Format(CoreResources.WhatIfShortCircuitingRuleMessageFormat, values);
+        }
+
+        override public IEnumerable<IDiagnostic> AnalyzeInternal(SemanticModel model, DiagnosticLevel diagnosticLevel)
+        {
+            foreach (var module in model.Root.ModuleDeclarations)
+            {
+                if (module.DeclaringSyntax is ModuleDeclarationSyntax moduleSyntax &&  module.TryGetSemanticModel().IsSuccess(out var moduleSemanticModel) && moduleSemanticModel is SemanticModel semanticModel)
+                {
+                    var moduleParamsPropertyObject = moduleSyntax.TryGetBody()?
+                                    .TryGetPropertyByName(LanguageConstants.ModuleParamsPropertyName);
+                    var moduleParams = (moduleParamsPropertyObject?.Value as ObjectSyntax)?.Properties;
+                    if (moduleParams == null)
+                    {
+                        continue;
+                    }
+
+                    var moduleParamsInput = new Dictionary<string, ITemplateLanguageExpression>();
+                    var moduleParamsHolder = new Dictionary<string, ObjectPropertySyntax>();
+                    foreach (var param in moduleParams)
+                    {
+                        moduleParamsInput.Add(param.Key.ToString(), new FunctionExpression("sentinel-placeholder", ImmutableArray.Create<ITemplateLanguageExpression>(new StringExpression(param.Key.ToString(), null, null, null)), ImmutableArray<ITemplateLanguageExpression>.Empty, null, null, null));
+                        moduleParamsHolder.Add(param.Key.ToString(), param);
+                    }
+
+                    var template = GetTemplate(semanticModel);
+                    TemplateWithParsedExpressions? templateResult;
+                    try
+                    {
+                        templateResult = TemplateEngine.ReduceTemplateLanguageExpressions(
+                            managementGroupName: null,
+                            subscriptionId: null,
+                            resourceGroupName: null,
+                            template: template,
+                            apiVersion: new StringExpression(NestedDeploymentResourceApiVersion, null, null, null),
+                            suppliedParameterValues: moduleParamsInput,
+                            parameterValuesPositionalMetadata: null,
+                            metadata: ImmutableDictionary<string, ITemplateLanguageExpression>.Empty,
+                            metricsRecorder: null,
+                            skipEvaluationOfNonDeterministicFunctionsForWhatIf: false);
+                    }
+                    catch (Exception ex)
+                    {
+                        Trace.WriteLine($"Exception occurred while reducing template language expressions: {ex}");
+                        continue;
```

Comment:
Should we emit a diagnostic here about analysis failing?

Reply:
Discussed offline - left a TODO to emit a diagnostic since it would likely cause duplicate diagnostics any time there is an error in the module param syntax.